ds-collector/TROUBLESHOOTING.md

```diff
@@ -0,0 +1,221 @@
+
+It is not a show-stopper to have `skipSudo=true` even in a Health Check context, but the preferable course of action here would be to configure `sudoers` to allow the `userName` to run `sudo` (without password) on each of the nodes, if only temporarily. This will result in the Health Check being able to assess a number of OS-level configuration settings.
+
+### Problems Connecting With `cqlsh`
+#### Condition:
+Failures connecting to `cqlsh` can manifest in a number of ways, but will typically have a command line `cqlsh` followed by `…` and some sort of error text, and the next line being `failed`. For example:
+
+    executing `cqlsh ds-collector-tests_cassandra-00_1 9042  --username=**** --password=**** -f /var/local/tmp/datastax/TEST_artifacts_2022_05_24_1513_1653405227/execute_schema.cql > schema.cql`… Connection error: ('Unable to connect to any servers', {'10.193.204.58': ConnectionShutdown('Connection to 10.193.204.58 was closed',)})
+    failed
+
+#### Why it Happens:
+The Collector attempts to run `cqlsh` from each Node, connecting to the local Node. It is unable to connect for some reason. Usually this is an unspecified username/password but some secured environments have configured for exclusive encryption access.
+
+#### What to do:
+Set username and password:
+- When connecting with `cqlsh`, do you need to specify `-u`? Configure `cqlshUsername=`.
+- When connecting with `cqlsh`, do you need to use a password? Configure `cqlshPassword=`.
+
+When the port is not 9042:
+- Configure `cqlsh_port=`
+
+When `$(hostname)` resolves to an address that is not being listened to for client connections (this is typically caused by ordering in `/etc/hosts`):
+- Configure `cqlsh_host=`. Note this could be a `$(command that gets the correct value)`.
+
+When Client Encryption is required:
+- Configure `cqlshSSL="true"`
+- You may need to configure `cqlsh_port=` to the secure port
+- You may need to specify additional command line options, for example `--cqlshrc=/path/to/cqhshrc`. Configure `cqlshOpts=`.
+
+Other command-line arguments are needed:
+- Configure `cqlshOpts=`
+
+### Error Involving `cpReadable.sh`
+#### Condition:
+
+    executing `/var/local/tmp/datastax/etc/cpReadable.sh /var/log/cassandra /var/local/tmp/datastax/TEST_artifacts_2022_05_24_1544_1653407055/logs > `…   usage: /var/local/tmp/datastax/etc/cpReadable.sh sourceDir targetDir
+
+    sourceDir : must be a directory containing files to copy
+    targetDir : must be an existing directory, or a valid path where a directory can be created
+
+    failed
+
+#### Why it Happens:
+Most commonly, it is because the `sourceDir` passed to `cpReadable.sh` (the first argument) is not a directory.
```

Comment:
this error makes it evident that the check that `logHome` is a valid directory needs to become a fail-fast check in the bash `node_checks()` function.

it's a one-liner check to add, and it will change how this problem item is written up here.